plugins/system76-launch/fu-system76-launch-device.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2021 Jeremy Soller <jeremy@system76.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-io-channel.h"
+#include "fu-system76-launch-device.h"
+
+#define SYSTEM76_LAUNCH_CMD_VERSION 3
+#define SYSTEM76_LAUNCH_CMD_RESET 6
+#define SYSTEM76_LAUNCH_TIMEOUT 1000
+
+struct _FuSystem76LaunchDevice {
+	FuUsbDevice			 parent_instance;
+};
+
+G_DEFINE_TYPE (FuSystem76LaunchDevice, fu_system76_launch_device, FU_TYPE_USB_DEVICE)
+
+static gboolean
+fu_system76_launch_device_setup (FuDevice *device, GError **error)
+{
+	GUsbDevice *usb_device = fu_usb_device_get_dev (FU_USB_DEVICE (device));
+	const guint8 ep_in = 0x82;
+	const guint8 ep_out = 0x03;
+	guint8 data[32] = { 0 };
+	gsize actual_len = 0;
+	g_autofree gchar *version = NULL;
+
+	/* this is the atmel bootloader */
+	fu_device_add_counterpart_guid (device, "USB\\VID_03EB&PID_2FF4");
+
+	/* Send version command */
+	data[0] = SYSTEM76_LAUNCH_CMD_VERSION;
+	if (!g_usb_device_interrupt_transfer (
+		usb_device,
+		ep_out,
+		data,
+		sizeof(data),
+		&actual_len,
+		SYSTEM76_LAUNCH_TIMEOUT,
+		NULL,
+		error
+	)) {
```

Comment:
can these please be aligned with the right hand bracket of g_usb_device_interrupt_transfer please.